### PR TITLE
#21574 Fix dispatcher use after system termination

### LINKED ISSUE #21574: Dangling threads in system dispatcher when dispatcher is used in future callback after termination
Reproducer:
```scala
import java.util.concurrent.Executors
import akka.actor.ActorSystem
import scala.concurrent.{ExecutionContext, Future}

def main(args: Array[String]): Unit = {
  val executor = Executors.newSingleThreadExecutor()
  val executionEC = ExecutionContext.fromExecutor(executor)
  val system = ActorSystem("system")
  val systemEC = system.dispatcher
  val longRunning = Future(Thread.sleep(5000))(executionEC)
  longRunning.onComplete(_ => println("something"))(systemEC)
  executor.shutdown()
  system.terminate()
}
```

After 5 seconds a new thread is started in the system dispatcher to execute the println in the callback. After callback has been executed that thread is not stopped.

A workaround is to stop the actorsystem from the callback.

### PR DESCRIPTION
**Pull request by 2m**
Fixes #21574

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

**Comment by jrudolph** [comment]
Would it be possible to provide a test for this behavior? It seemed we had a reproducer for the issue.

**Comment by patriknw** [comment]
+1 for test. It should be possible to count number of threads before and after to verify that everything have been shutdown.

**Comment by 2m** [comment]
Added a test and amended the comment.

File: akka-actor-tests/src/test/scala/akka/dispatch/DispatcherShutdownSpec.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        .filter(_.startsWith("DispatcherShutdownSpec-akka.actor.default"))
+        .size
+
+      val system = ActorSystem("DispatcherShutdownSpec")
+      threadCount should be > 0
+
+      Await.ready(system.terminate(), 1.second)
+      Await.ready(Future(akka.Done)(system.dispatcher), 1.second)
+
+      TestKit.awaitCond(threadCount == 0, 1.second)
```
Comment (johanandren) [inline akka-actor-tests/src/test/scala/akka/dispatch/DispatcherShutdownSpec.scala]: isn't 1 second a bit short for all these? what about using `remainingOrDefault`?

**Review by johanandren — APPROVED** [review]
LGTM but I think timeouts are too short.

File: akka-actor-tests/src/test/scala/akka/dispatch/DispatcherShutdownSpec.scala (hunk 1 above)
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/dispatch/DispatcherShutdownSpec.scala]: this is not a TestKit spec, so using `3.seconds` would be good

**Review by patriknw — APPROVED** [review]
LGTM, after changing test timeouts

**Comment by johanandren** [comment]
PLS BUILD

**Comment by ktoso** [comment]
LGTM

**Pull request merged**